libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tooling/BuildKotlinToolingMetadataTask.kt

```diff
@@ -55,9 +54,11 @@ abstract class BuildKotlinToolingMetadataTask : DefaultTask() {
         const val defaultTaskName: String = "buildKotlinToolingMetadata"
     }
 
+    @get:OutputDirectory
+    val outputDirectory: File = project.buildDir.resolve("kotlinToolingMetadata")
+
     @get:OutputFile
-    val outputFile: Property<File> = project.objects.property(File::class.java)
-        .convention(project.buildDir.resolve("kotlinToolingMetadata").resolve("kotlin-tooling-metadata.json"))
+    val outputFile: File = outputDirectory.resolve("kotlin-tooling-metadata.json")
```

Comment:
Is is redundant to have both file and its parent dir as outputs, can you remove `outputFile`?

Reply:
Yes! And since we are including everything located in this directory, I am convinced that it is not a bad idea to also clean the directory if the building task is executed. 


See: https://github.com/JetBrains/kotlin/pull/4530/commits/dec813ede64346ae20af46de86b25f39e6d16336